homeassistant/components/tvoverlay/config_flow.py

```diff
@@ -0,0 +1,79 @@
+"""Config flow for TvOverlay integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from tvoverlay import Notifications
+from tvoverlay.exceptions import ConnectError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TvOverlayFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for TvOverlay."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is not None:
+            self._async_abort_entries_match(
+                {CONF_HOST: user_input[CONF_HOST], CONF_NAME: user_input[CONF_NAME]}
+            )
+            error, info = await self._async_try_connect(user_input[CONF_HOST])
+            if error is not None:
+                errors["base"] = error
+            elif info is not None:
+                device_name = (
+                    info.get("result", {})
+                    .get("settings", {})
+                    .get("deviceName", DEFAULT_NAME)
+                )
+                device_id = (
+                    info.get("result", {}).get("status", {}).get("id", DEFAULT_NAME)
+                )
```

Comment:
I don't think this looks good, why can't we trust what we received from the app? Also, a default name is absolutely unacceptable as a unique_id.